pkg/providers/imagefamily/nodeimage_test.go

```diff
@@ -171,39 +183,126 @@ var _ = Describe("NodeImageProvider tests", func() {
 			ctx = options.ToContext(ctx, testOptions)
 		})
 
-		It("should match expected images for Ubuntu2204", func() {
-			foundImages, err := nodeImageProvider.List(ctx, nodeClass)
-			Expect(err).ToNot(HaveOccurred())
-			Expect(foundImages).To(ContainElements(getExpectedTestSIGImages(*nodeClass.Spec.ImageFamily, sigImageVersion, kubernetesVersion)))
-		})
+		Context("List FIPS Images When FIPSMode Is Explicitly FIPS", func() {
+			BeforeEach(func() {
+				nodeClass.Spec.FIPSMode = lo.ToPtr(string(v1beta1.FIPSEnabled))
+			})
 
-		// This test changes depending on the Kubernetes version, in effect making the following version-specific tests unnecessary.
-		// They are still kept for clarity and to ensure that the behavior is explicitly tested.
-		It("should match expected images for AzureLinux, depending on the Kubernetes version", func() {
-			nodeClass.Spec.ImageFamily = lo.ToPtr(v1beta1.AzureLinuxImageFamily)
+			It("should match expected images for FIPS Ubuntu2004", func() {
```

Comment:
nit: naming I'd adjust as the selection of `Ubuntu2004` happens behind the scenes and might shift in the future.
```suggestion
			It("should match expected images for FIPS when using generic Ubuntu", func() {
```